ui-modules/home/app/views/about/about.controller.js

```diff
@@ -55,17 +61,141 @@ export function aboutStateConfig($stateProvider) {
+                backdrop: 'static',
+                windowClass: 'quick-launch-modal',
+                size: 'md',
+                resolve: {
+                    node: function () {
+                        return states.nodes[states.ownId];
+                    }
+                }
+            });
+            modalInstance.result.then(
+                (promiseList) => {
+                    $scope.template = 'spinnerTemplate';
+                    Promise.allSettled(promiseList).then((values) => {
+                        let event = new CustomEvent('update-states', {});
+                        container.dispatchEvent(event);
+                    });
+                    modalInstance = null;
+                },
+                () => {
+                    $scope.template = 'spinnerTemplate';
+                    let event = new CustomEvent('update-states', {});
+                    container.dispatchEvent(event);
+                    modalInstance = null;
+                }
+            );
+        }
+
+    };
+
+    function nodeManagementModalController($scope, $uibModalInstance, node) {
+
+        // newStatus and newPriority are required to belong to the instance of the modal
+        let vm = this;
+        vm.newPriority = node.priority;
+        vm.newStatus = node.status;
+        
+        $scope.node = node;
+        $scope.statuses = ["MASTER", "STANDBY", "HOT_STANDBY", "HOT_BACKUP"];
+        $scope.now = Date.now();
+        $scope.showEditOptions = false;
+
+        $scope.applyChangesAndQuit = function () {
+            let promiseList = [];
+            if ($scope.node.priority !== vm.newPriority) {
+                let result = serverApi.setHaPriority(vm.newPriority);
+                promiseList.push(result);
+            }
+            if ($scope.node.status !== vm.newStatus) {
+                let result = serverApi.setHaStatus(vm.newStatus);
+                promiseList.push(result);
+            }
+            $uibModalInstance.close(promiseList);
+
+        }
+
+        $scope.cancelAndQuit = function () {
+            vm.newPriority = $scope.node.priority;
+            vm.newStatus = $scope.node.status;
+            $uibModalInstance.dismiss();
+        }
+
+        $scope.doShowEditOptions = function () {
+            $scope.showEditOptions = true;
+        }
+
+    }
+
+    $scope.removeNode = function (nodeId) {
+        $scope.template = 'spinnerTemplate';
+        let removeNode = serverApi.removeHaTerminatedNode(nodeId);
+        removeNode.then(data => {
+            $scope.expectedNodeCounter--;
+            let event = new CustomEvent('update-states', {});
+            $scope.container.dispatchEvent(event);
+        });
+    }
+
+    $scope.removeAllTerminatedNodes = function () {
+        $scope.template = 'spinnerTemplate';
+        let removeNodes = serverApi.removeHaTerminatedNodes();
+        removeNodes.then(data => {
+            for (const node in $scope.states.nodes) {
```

Comment:
consider using `$scope.states.nodes.forEach(node => { logic here )` instead for a better syntax